sequencer/src/persistence/sql.rs

```diff
@@ -277,13 +404,16 @@ impl From<PruningOptions> for PrunerCfg {
 impl PersistenceOptions for Options {
     type Persistence = Persistence;
 
-    async fn create(self) -> anyhow::Result<Persistence> {
+    async fn create(mut self) -> anyhow::Result<(Self, Self::Persistence)> {
```

Comment:
Would it be cleaner to have this take `&mut self` and only return `Self::Persistence`?